[ews-build] Bring up Safer-CPP-Checks queue!

briannafan opened the pull request:
#### 17a253991b6e47ca9fdb018f263363dbd7362cf6
<pre>
[ews-build] Bring up Safer-CPP-Checks queue!
<a href="https://bugs.webkit.org/show_bug.cgi?id=280636">https://bugs.webkit.org/show_bug.cgi?id=280636</a>
<a href="https://rdar.apple.com/136993599">rdar://136993599</a>

Reviewed by Aakash Jain and Ryan Haddad.

Config change to bring up Mac-Safer-CPP-Checks EWS queue.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble):
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR): Add &apos;mac-safer-cpp&apos; to safe-merge-queue checks.

Canonical link: <a href="https://commits.webkit.org/284530@main">https://commits.webkit.org/284530@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b7f77a94cb828ab5b45f6f5440c22b3fb8fae20

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69687 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49087 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22438 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73772 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20845 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71804 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56888 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20696 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/55377 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13835 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72753 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44749 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60102 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35857 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41415 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17540 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19222 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63351 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17885 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75485 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13669 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17143 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/63050 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/69365 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13709 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60185 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/62966 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10992 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4578 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10640 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44891 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45965 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/47229 "Build was cancelled. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45706 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -287,6 +293,12 @@
       "workernames": ["ews181", "ews182"]
     },
+    {
+      "name": "macOS-Safer-CPP-Checks-EWS", "shortname": "mac-safer-cpp", "icon": "buildAndTest",
```
Please verify whether `testOnly` icon would be more appropriate for this queue.

briannafan replied:
Upon discussion with @ryanhaddad, `buildOnly` is actually the best option, since it's a custom build process rather than a test suite that's being run.

aj062 commented on Tools/CISupport/ews-app/ews/common/github.py:
```diff
@@ -213,7 +213,7 @@ class GitHubEWS(GitHub):
                           ['services', 'vision-sim', 'mac-wk2-stress', 'api-gtk', ''],
                           ['merge', 'vision-wk2', 'mac-intel-wk2', 'jsc-armv7', ''],
-                          ['unsafe-merge', 'tv', '', 'jsc-armv7-tests', ''],
+                          ['unsafe-merge', 'tv', 'mac-safer-cpp', 'jsc-armv7-tests', ''],
```
Would be a good idea to add the status-bubble in a separate PR which would land little later, once the queue is running properly in Production (without any issues like false positives). Meanwhile, make sure that this queue doesn't block/comment on the PRs.
Especially important for this queue, since this queue is expected to directly impact the engineers and they would have to learn some new ways of resolving issues found by ews queue.

However, if adequate testing has been done, it's fine to land all these changes together.

briannafan replied:
This queue has been running on dev for a couple weeks and the results have been monitored and shared with the engineers working on the checker. I think we have enough testing that we can land all the changes together and comment on PRs.

aj062 commented on Tools/CISupport/ews-app/ews/common/buildbot.py:
```diff
@@ -50,6 +50,7 @@ class Buildbot():
         'mac-wk2': 'mac',
         'mac-intel-wk2': 'mac',
         'mac-wk2-stress': 'mac',
+        'mac-safer-cpp': 'mac',
```
This seems incorrect. This queue doesn't seems to be triggered by macOS queue (e.g.: `macOS-Ventura-Release-Build-EWS`).

briannafan replied:
Good catch. I was working on autopilot and forgot this queue doesn't have a trigger.

aj062 commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -287,6 +293,12 @@
       "workernames": ["ews181", "ews182"]
     },
+    {
+      "name": "macOS-Safer-CPP-Checks-EWS", "shortname": "mac-safer-cpp", "icon": "buildAndTest",
```
please verify if we should have macOS in this name of this queue. I wonder if we might have similar queue for iOS or other platforms in future.

briannafan replied:
Yes, we might have this on iOS in the future.

briannafan committed:
[ews-build] Bring up Safer-CPP-Checks queue!
https://bugs.webkit.org/show_bug.cgi?id=280636
rdar://136993599

Reviewed by Aakash Jain and Ryan Haddad.

Config change to bring up Mac-Safer-CPP-Checks EWS queue.

* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble):
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(CheckStatusOfPR): Add 'mac-safer-cpp' to safe-merge-queue checks.

Canonical link: https://commits.webkit.org/284530@main
